src/dmd/cpreprocess.d

```diff
@@ -61,16 +61,17 @@ FileName preprocess(FileName csrcfile, out bool ifile)
     }
     else version (Windows)
     {
-        /* This actually works, but fails the test suite because:
-           1. For Microsoft's preprocessor, it insists on printing the filename it's preprocessing,
-              which causes all the fail_compilation tests to fail. Yes, even with /nologo.
-              cl.exe does not appear to have a "quiet" mode.
-           2. For Win32 with Digital Mars, the tests all fail because sppn.exe, the DMC preprocessor,
-              is not on the path. Even if it was, it would still fail because the DMC headers are not
-              there. To get the headers and sppn.exe, dmc will need to be installed on the test machines:
-              http://ftp.digitalmars.com/Digital_Mars_C++/Patch/dm857c.zip
+        /*
+           To get sppn.exe: http://ftp.digitalmars.com/sppn.zip
+           To get the dmc C headers, dmc will need to be installed:
+           http://ftp.digitalmars.com/Digital_Mars_C++/Patch/dm857c.zip
          */
-        if (target.objectFormat() == Target.ObjectFormat.coff)
+        const name = FileName.name(csrcfile.toString());
+        const ext = FileName.ext(name);
+        assert(ext);
+        const ifilename = FileName.addExt(name[0 .. name.length - (ext.length + 1)], i_ext);
+        auto status = runPreprocessor("cl /P", csrcfile.toString(), ifilename);
+        if (status)
```

Comment:
Rebasing seems to have lost some changes here (and added duplicate execution of the preprocessor).

When adding back `sppn`, you should now call `cppCommamd()` instead of using an immediate literal, and add it there as a default for OMF.

The test suite should then find the preprocessor if you add
```
CPPCMD=$PWD/dm/bin/sppn.exe
```
here: https://github.com/dlang/dmd/blob/master/.azure-pipelines/windows.sh#L34

Reply:
Klasse!